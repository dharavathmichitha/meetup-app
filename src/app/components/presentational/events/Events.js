import React from 'react';
import PropTypes from 'prop-types';
import { getDate } from '../../../utils/date-utils';
import EventsPerDay from './EventsPerDay';

const Events = ({ eventsListData, cssClass }) => {
  const getEventPerDay = (eventsList, index) => {
    const dayHasEvents = eventsList.length !== 0;
    const date = dayHasEvents && getDate(eventsList[0].startDate);
    const key = `events-wrapper-${index}`;
    return (
      <div key={key} className="events-wrapper" role="list">
        <EventsPerDay date={date} events={eventsList} />
      </div>
    );
  };

  const shouldDisplayEventsPerDay = eventsListData.length !== 0;
  const eventsPerDay = shouldDisplayEventsPerDay ? (
    eventsListData.map(getEventPerDay)
  ) : (
    <div className="events--fallback-text">There are no avaliable events... :(</div>
  );

  return (
    <>
      <section className={`events ${cssClass}`}>{eventsPerDay}</section>
    </>
  );
};

Events.propTypes = {
  cssClass: PropTypes.string.isRequired,
  eventsListData: PropTypes.arrayOf(
    PropTypes.arrayOf(
      PropTypes.shape({
        id: PropTypes.number.isRequired,
        isFree: PropTypes.bool.isRequired,
        name: PropTypes.string.isRequired,
        city: PropTypes.number.isRequired,
        startDate: PropTypes.string.isRequired,
        endDate: PropTypes.string.isRequired,
      }).isRequired,
    ),
  ).isRequired,
};

export default Events;
