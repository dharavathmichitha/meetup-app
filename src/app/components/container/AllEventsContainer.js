import React, { useEffect, useState, useContext } from 'react';
import PropTypes from 'prop-types';
import EventSections from '../presentational/events/EventSections';
import { getEventsByDate, getEventsByName } from '../../api/events';
import { SearchContext, FreeEventsContext } from '../../utils/hooks-utils';

const initEventsPerDay = [];

export default function EventSectionContainer() {
  const [eventsListData, setEventsPerDay] = useState(initEventsPerDay);
  const { searchValue } = useContext(SearchContext);
  const { freeEvents } = useContext(FreeEventsContext);

  useEffect(() => {
    const eventName = searchValue;
    const validationRegex = RegExp("^[a-zA-Z0-9',:]+( [a-zA-Z0-9',:]+)*$");
    const isValidInput = validationRegex.test(eventName);

    if (isValidInput) {
      getEventsByName(setEventsPerDay, eventName);
    } else {
      getEventsByDate(setEventsPerDay);
    }
  }, [searchValue]);

  useEffect(() => {
    setEventsPerDay(freeEvents);
  }, [freeEvents]);

  return <EventSections eventsListData={eventsListData} />;
}

EventSectionContainer.prototype = {
  eventsListData: PropTypes.array.isRequired,
};
