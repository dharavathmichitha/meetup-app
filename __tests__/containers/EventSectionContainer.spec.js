import React from 'react';
import { mount } from 'enzyme';
import Events from '../../src/app/components/presentational/events/Events';
import AllEventsContainer from '../../src/app/components/container/AllEventsContainer';

beforeEach(() => {
  jest.resetModules();
});

describe('AllEventsContainer', () => {
  it('should render container without events included', () => {
    const wrapper = mount(<AllEventsContainer />);
    expect(wrapper.contains(Events)).toEqual(true);
    expect(wrapper.html()).toMatchSnapshot();
  });
});
