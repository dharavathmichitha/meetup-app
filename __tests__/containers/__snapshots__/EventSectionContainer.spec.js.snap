// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AllEventsContainer should render container without events included 1`] = `"<section class=\\"filters\\"><input aria-label=\\"Close\\" id=\\"searchinput\\" type=\\"text\\" class=\\"filters__event-search\\" placeholder=\\"Search an event...\\" value=\\"\\"><button type=\\"button\\" class=\\"filters__free-events btn btn\\">Discover Free Events</button></section><section class=\\"events \\"></section>"`;
